module/apmsql/conn_test.go

```diff
@@ -0,0 +1,31 @@
+package apmsql_test
+
+import (
+	"context"
+	"database/sql/driver"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"go.elastic.co/apm/module/apmsql"
+	_ "go.elastic.co/apm/module/apmsql/sqlite3"
+)
+
+func TestConnUnwrap(t *testing.T) {
+	db, err := apmsql.Open("sqlite3", ":memory:")
+	require.NoError(t, err)
+	defer db.Close()
+
+	type Unwrapper interface {
+		Unwrap() driver.Conn
+	}
+
+	ctx := context.Background()
+	conn, err := db.Conn(ctx)
+	_ = conn.Raw(func(driverConn interface{}) error {
+		unwrappedConn := driverConn.(Unwrapper)
+		require.Equal(t, "*apmsql.connBeginTx", fmt.Sprintf("%T", unwrappedConn))
```

Comment:
This is just type-asserting - should the unwrapper function be called, to get the underlying driver.Conn? And change the assertion below to check that the result is the sqlite3 Conn implementation?

Reply:
Hi @axw , you're right. I've improved the assertion to verify against `*sqlite3.SQLiteConn`.

Reply:
Hi @axw , you're right. I've improved the assertion to verify against `*sqlite3.SQLiteConn`.